terraform/python/ec2/main.tf

```diff
@@ -0,0 +1,236 @@
+}
+
+resource "aws_instance" "remote_service_instance" {
+  ami                                   = data.aws_ami.ami.id # Amazon Linux 2 (free tier)
+  instance_type                         = "t3.small"
+  key_name                              = local.ssh_key_name
+  iam_instance_profile                  = "APP_SIGNALS_EC2_TEST_ROLE"
+  vpc_security_group_ids                = [aws_default_vpc.default.default_security_group_id]
+  associate_public_ip_address           = true
+  instance_initiated_shutdown_behavior  = "terminate"
+  metadata_options {
+    http_tokens = "required"
+  }
+
+  tags = {
+    Name = "remote-service-${var.test_id}"
+  }
+}
+
+resource "null_resource" "remote_service_setup" {
+  connection {
+    type = "ssh"
+    user = var.user
+    private_key = local.private_key_content
+    host = aws_instance.remote_service_instance.public_ip
+  }
+
+  provisioner "remote-exec" {
+    inline = [
+      # Install Python and wget
+      "sudo yum install wget -y",
+      "sudo yum install unzip -y",
+      "sudo dnf install -y python3.9",
+      "sudo dnf install -y python3.9-pip",
+
+      # Copy in CW Agent configuration
+      "agent_config='${replace(replace(file("./amazon-cloudwatch-agent.json"), "/\\s+/", ""), "$REGION", var.aws_region)}'",
+      "echo $agent_config > amazon-cloudwatch-agent.json",
+
+      # Get and run CW agent rpm
+      "wget -O cw-agent.rpm https://amazoncloudwatch-agent-us-east-1.s3.us-east-1.amazonaws.com/amazon_linux/amd64/1.300031.0b313/amazon-cloudwatch-agent.rpm",
+      "sudo rpm -U ./cw-agent.rpm",
+      "sudo /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a fetch-config -m ec2 -s -c file:./amazon-cloudwatch-agent.json",
+
+      # Get ADOT Wheel and install it
+
+      "aws s3 cp s3://aws-appsignals-sample-app-prod-${var.aws_region}/${var.adot_wheel_name} ./${var.adot_wheel_name}",
+      "python3.9 -m pip install ${var.adot_wheel_name}",
+
+      # Get and run the sample application with configuration
+      "aws s3 cp ${var.sample_app_zip} ./python-sample-app.zip",
+      "unzip -o python-sample-app.zip",
+
+      # Export environment variables for instrumentation
+      "cd ./django_remote_service",
+      "export DJANGO_SETTINGS_MODULE=\"django_remote_service.settings\"",
+      "python3.9 -m pip install -r requirements.txt --force-reinstall",
+      "export OTEL_PYTHON_DISTRO=\"aws_distro\"",
+      "export OTEL_PYTHON_CONFIGURATOR=\"aws_configurator\"",
+      "export OTEL_METRICS_EXPORTER=none",
+      "export OTEL_TRACES_EXPORTER=otlp",
+      "export OTEL_AWS_APP_SIGNALS_ENABLED=true",
+      "export OTEL_AWS_APP_SIGNALS_EXPORTER_ENDPOINT=http://localhost:4315",
+      "export OTEL_EXPORTER_OTLP_TRACES_ENDPOINT=http://localhost:4315",
+      "export OTEL_EXPORTER_OTLP_TRACES_PROTOCOL=grpc",
+      "export OTEL_EXPORTER_OTLP_METRICS_PROTOCOL=grpc",
+      "export OTEL_SERVICE_NAME=sample-remote-application-${var.test_id}",
```

Comment:
same here. make it `python-sample-remote-application-...`.

Reply:
Updated